crates/node/src/main.rs

```diff
@@ -366,8 +375,52 @@ async fn p2p_beacon(config: P2PBeaconConfig) -> Result<()> {
     let p2p_addr = p2p.node().start_listening().await?;
     tracing::info!("listening for p2p at {}", p2p_addr);
 
+    let mut connected_nodes = 0;
+    let mut try_count = 0;
+
+    while connected_nodes == 0 && try_count < config.cluster_join_attempt_limit {
+        for addr in config.p2p_discovery_addrs.clone() {
+            tracing::info!("connecting to p2p peer {}", addr);
+            match addr.to_socket_addrs() {
+                Ok(mut socket_iter) => {
+                    if let Some(peer) = socket_iter.next() {
+                        let (connected, fail) = p2p.do_connect(peer, true).await;
+                        connected_nodes += connected.len();
+                        if !fail.is_empty() {
+                            tracing::info!("Peer connection, fail to connect to these peers:{fail:?}");
+                        }
+                    }
+                }
+                Err(err) => {
+                    tracing::error!("failed to resolve {}: {}", addr, err);
+                }
+            }
+        }
+
+        try_count += 1;
+    }
+
+    if !config.p2p_discovery_addrs.is_empty() && connected_nodes == 0 {
+        tracing::info!("No discovery addresses configured or none could be resolved. Assuming we're the first node.");
+    }
+
+    // Start a basic healthcheck so kubernetes has something to wait on.
+    let listener = TcpListener::bind(config.http_healthcheck_listen_addr).await?;
+    tracing::info!("Healthcheck listening on: {}", listener.local_addr()?);
+
     loop {
-        sleep(Duration::from_secs(1));
+        let (stream, _) = listener.accept().await?;
+
+        let io = TokioIo::new(stream);
+
+        tokio::task::spawn(async move {
+            if let Err(err) = http1::Builder::new()
+                .serve_connection(io, service_fn(ok))
+                .await
+            {
+                tracing::error!("Error serving connection: {:?}", err);
+            }
+        });
     }
```

Comment:
To avoid starting a new http server you can move the lines 243,244:
```
       let scheduler_watchdog_sender =
        watchdog::start_healthcheck(config.http_healthcheck_listen_addr).await?;
```
Just before the `if` for example line 239 and remove these.
It will activate the health check watchdog for all type of node. The scheduler verification will fail, but the / entry point will be ok.
I was thinking it will log an error. If it's too annoying, keep the http server you add, and I'll remove it when I'll change the watchdog to activate per service.

Reply:
I wasn't sure if it was safe to re-use that so I just whipped up something quick to get me past the issue I had (without the readiness check kubernetes won't wait to start the next statefulset pod). I'm ok with ignoring any errors it logs for now so fix it when you get around to it.